### ISSUE TITLE
Syntax diagram extends past right edge of page in PDF built by DITA OT

### ISSUE THREAD

**Issue by ghannington-Rocket** [robander/svg-syntaxdiagrams#15]
I'm using DITA OT 3.5.2 with the svg-syntaxdiagrams 2.0 plugins.

DITA:

````XML
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE reference PUBLIC "-//OASIS//DTD DITA Reference//EN" "reference.dtd">
<reference id="syntax" xml:lang="en">
    <title>Syntax diagram</title>
    <refbody>
        <refsyn>
            <syntaxdiagram>
                <groupseq>
                    <kwd>COMMAND</kwd>
                </groupseq>
                <groupchoice importance="optional" id="syntax-group">
                    <groupcomp importance="default">
                        <kwd>PARAMETER</kwd>
                        <delim>(</delim>
                        <kwd>CHOICE0</kwd>
                        <delim>)</delim>
                    </groupcomp>
                    <groupcomp>
                        <kwd>PARAMETER</kwd>
                        <delim>(</delim>
                        <groupchoice>
                            <kwd>CHOICE1</kwd>
                            <kwd>CHOICE2</kwd>
                            <var>variable</var>
                        </groupchoice>
                        <delim>)</delim>
                    </groupcomp>
                </groupchoice>
                <groupchoice conref="#syntax/syntax-group"/>
                <groupchoice conref="#syntax/syntax-group"/>
                <groupchoice conref="#syntax/syntax-group"/>
                <groupchoice conref="#syntax/syntax-group"/>
            </syntaxdiagram>
        </refsyn>
    </refbody>
</reference>
````

Command to format the DITA to PDF:

````
dita --input=syntax.dita --format=pdf
````

Resulting PDF:

[syntax.pdf](https://github.com/robander/svg-syntaxdiagrams/files/5063391/syntax.pdf)

How do I prevent this?

(By the way: thank you so much for the "cleaner" rendering, sans boxes, and with fewer arrowheads.)

**Comment by ghannington-Rocket** [comment]
Argh! I inadvertently submitted this issue while logged in using a GitHub ID that I'm supposed to be reserving for a particular use (not this one!). But it seems I can't delete an issue after I've created it. Ah well. I'll try to be more careful in future.

**Comment by GrahamHannington** [comment]
Similar to issue #16, I wondered if this problem was introduced by a change in DITA OT, so I tried the same thing with DITA OT 3.3.4 (the version cited in the readme of this repo).

Nope. Same problem 🙁.

**Comment by GrahamHannington** [comment]
Trying to figure this out by myself... I'd appreciate advice... how do the svg-syntaxdiagrams plugins decide the diagram width? Specifically, for PDF output (although, I'm also interested in the same question for HTML, especially if the answer is different).

I've searched the code for "width", and I'm examining the results, but, while I can write XSLT, XSL-FO, SVG, and CSS, I'm a newbie to this code: I don't really understand what I'm looking at.

At first glance, this looks promising, in `com.moldflow.dita.syntaxdiagram2svg/resource/constantsPDF.xml`:

````xml
    <!-- Maximum diagram size, in SVG pixels. -->
<!-- RDA: change from 650 to 500 so this fits on a PDF page -->
    <scalar name="diagram_max_width">500</scalar>
````

but—finger in the air, not really knowing what I'm talking about—what I'm seeing in the PDF already looks wider than any 500 pixels (SVG or otherwise) I've seen in the last 30 years.

Still, I'm going to try tweaking it, and then move on to the next possibility; although, frankly, I have limited time to look at this before I have to give up.

Any help anyone could offer would be greatly appreciated.

**Comment by GrahamHannington** [comment]
Ah. Now I'm getting somewhere.

Reducing `diagram_max_width` and the—I imagine, related—`diagram_wrap_width` in `constantsPDF.xml` to 200 and 250, respectively, had no visible effect on the PDF output.

However. setting those same values in `constants.xml` _did_ have the desired effect.

I deliberately went in hard and made a big change; the diagrams are now _too_ narrow: they wrap sooner than they need to. So, I'll ease back, and see if I can figure out how to specify the exact appropriate width, in SVG pixels, for, say, the default PDF output from DITA OT. Perhaps that's exactly what's in `constantsPDF.xml`! Perhaps that's why it's there. Perhaps I should just try renaming those two files. Perhaps this is already documented somewhere in the plugins folders, and I'll look back at the comment I'm writing now and curse myself. Still, I often find it useful to log the steps of my research as I'm performing it, rather than waiting until I've reached the final solution, and just document that. With apologies to anyone reading this and rolling their eyes at my newbieness.

While I don't believe in magic, and without having much of a clue as to how the svg-syntaxdiagrams plugins might do this, I had imagined that the plugin(s) would somehow be contextually aware of the available width.

**Comment by GrahamHannington** [comment]
Yep. Just renaming `constants.xml` to something else, and `constantsPDF.xml` to `constants.xml`, did the trick. Although, that tells me that I need to tweak those values according to my particular needs: for example, PDF or HTML, and the widths I want in each of those. Is there a way to set those values without editing those XML files?

The following strikes me as downright odd (obviously, I'm missing something): while the example syntax diagram markup I've presented here, in this issue, now renders correctly, the example diagram with fragments in issue #16 continues to render too wide, in addition to the fragref-specific problems I describe in that issue. Why don't these width values appear to affect the syntax diagram with fragments? Do syntax diagrams with fragments have their own settings?

**Comment by robander** [comment]
As you know, these plugins were used in IBM as part of a large collection of plugins used to create styled PDF. It's been a long time since I've looked at this code at all, and I no longer have access to those plugins. From my very vague memories, I think the IBM PDF code (outside of the syntax code) must have set this property to use the PDF copy of the constants:
https://github.com/robander/svg-syntaxdiagrams/blob/master/com.moldflow.dita.syntaxdiagram2svg/syntaxdiagram2svg.xml#L21

That property determines which files get used, and is the one you'll want to set if you're using the PDF constants (or your own constants).

I can't help with the 200 vs 500 issue -- my only guess is that you might be using FOP, which I know scaled many of these diagrams up far beyond their original size. The pixel width should be respected with licensed renderers.
